result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -28,30 +28,57 @@ fun <V: Any?, E: Exception> Result<V, E>.getOrNull(): V? {
     }
 }
 
-inline fun <V : Any?, U : Any?, E : Exception> Result<V, E>.map(transform: (V) -> U): Result<U, E> = try {
+inline infix fun <V : Any?, E : Exception> Result<V, E>.getExceptionOrElse(fallback: (V) -> E): E {
+    return when (this) {
+        is Result.Success -> fallback(value)
+        is Result.Failure -> error
+    }
+}
+
+fun <V : Any?, E : Exception> Result<V, E>.getExceptionOrNull(): E? {
+    return when (this) {
+        is Result.Success -> null
+        is Result.Failure -> error
+    }
+}
+
+inline fun <V : Any?, E : Exception, U : Any?, F : Exception> Result<V, E>.mapEither(success: (V) -> U, failure: (E) -> F): Result<U, F> {
```

Comment:
In the previous implementation, we use the naming `fold` for this right? Do you think the old naming was not descriptive enough? 🤔

Reply:
There is a difference, `mapEither` returns another `Result`, so it is basically a shortening for having to chain a `.map` and a `.flatmap`

Reply:
OK!